Dockerfile

```diff
@@ -5,26 +5,28 @@ ENV DEFAULT_KERNEL=ir \
     TINI_VERSION=v0.16.1 \
     CIVIS_JUPYTER_NOTEBOOK_VERSION=1.0.1
 
-RUN DEBIAN_FRONTEND=noninteractive apt-get update -y --no-install-recommends && \
+# for python3.7
+RUN echo 'deb http://ftp.debian.org/debian stable main' >> /etc/apt/sources.list
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -y  && \
     apt-get install -y --no-install-recommends \
-        curl \
         wget \
         fonts-dejavu \
         gfortran \
-        python-dev \
         vim \
         nano \
         emacs \
-        gcc &&  \
-    apt-get clean -y && \
-    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-
-# Install pip
-RUN curl -s https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
-    python get-pip.py && \
-    pip install --upgrade pip setuptools && \
-    rm -rf ~/.cache/pip && \
-    rm -f get-pip.py
+        gcc  \
+        build-essential \
+        python3.7 \
+        python3-pip \
+        python3-setuptools \
+        libcurl3 && \
+        apt-get clean -y && \
+        rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# instead of virtual env, just use python3.7 as default
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.7 1
```

Comment:
A little worried about this breaking things (https://www.debian.org/doc/packaging-manuals/python-policy/ch-python.html), though I suppose no system stuff should be being installed/messed with from within a notebook. Any reason for wanting to avoid venv?

Reply:
> though I suppose no system stuff should be being installed/messed with from within a notebook

This was my thought.

It's a little easier on the R side if the default python is python3 rather than using a virtual environment. R's handling/awareness of the python virtual environment variables is hit or miss, and it's a bit clunky to get working occasionally. I did have a working version, but this was simpler.